mlflow/tracking/client.py

```diff
@@ -493,17 +493,21 @@ def update_model_version(self, name, version, description=None):
                                                                 description=description)
 
     @experimental
-    def transition_model_version_stage(self, name, version, stage):
+    def transition_model_version_stage(self, name, version, stage, archive_existing_versions=False):
         """
         Update model version stage.
 
         :param name: Registered model name.
         :param version: Registered model version.
         :param stage: New desired stage for this model version.
+        :param archive_existing_versions: If this flag is set, all existing model
+            versions in the stage will be atomically moved to the "archived" stage.
```

Comment:
I think you should mention that this should only be applied for staging and production stages otherwise it'll throw an error.